clients/src/main/java/org/apache/kafka/common/metrics/Sensor.java

```diff
@@ -223,8 +229,14 @@ public void checkQuotas(long timeMs) {
                 Quota quota = config.quota();
                 if (quota != null) {
                     double value = metric.measurableValue(timeMs);
-                    if (!quota.acceptable(value)) {
-                        throw new QuotaViolationException(metric, value, quota.bound());
+                    if (metric.measurable() instanceof TokenBucket) {
+                        if (value <= 0) {
```

Comment:
we probably just want `value < 0` check here, right? Otherwise the throttle time will be 0 anyways.

Reply:
Yes, that makes sense.